### DISCUSSION TITLE
Source for METAR/TAF

### DISCUSSION THREAD

**Discussion by drmrbrewer** [aeharding/metar-taf-parser#60] [General]
I'm curious to know what you consider to be the best source for METAR/TAF.  I have been using [NOAA](https://www.aviationweather.gov/adds/dataserver/) for years, but it isn't the most reliable, and also seems somewhat dated nowadays.  More recently I've come across [this source](https://www.checkwxapi.com/) which seems pretty good.

**Comment by aeharding** [comment]
aviationweather.gov works fine (also note there's beta.aviationweather.gov e.g. https://beta.aviationweather.gov/cgi-bin/data/metar.php?ids=KORD&format=json with an updated JSON api, although it parses the report which is unecessary overhead if you're already using metar-taf-parser).

You can also query directly from the source e.g. https://tgftp.nws.noaa.gov/data/forecasts/taf/stations/KMSN.TXT but that doesn't have airport metadata available (coordinates, elevation etc). That's what the example site uses.

I'm pretty sure tgftp.nws.noaa.gov is the most up to date, since I believe (don't quote me) this is where the METAR/TAF reports are directly uploaded from the stations themselves.

**Reply by aeharding** [reply]
Example site parsing here: https://github.com/aeharding/metar-taf-parser/blob/main/example/src/services/noaa.ts

**Reply by drmrbrewer** [reply]
Thanks for pointing out the beta site.  Looks potentially better, though at present I often make use of a lookup based on lat/long rather than station ID directly... the [beta docs](https://beta.aviationweather.gov/data/example/) suggest that "Multiple stations, bounding box" comes close, but it's not apparent that the list returned is sorted by distance from the requested point (so I can just grab the first)... probably not.  It also just returns the raw METAR string, which is OK if combining it with your library to do the parsing.  At present I'm using the [radial distance](https://www.aviationweather.gov/dataserver/example?datatype=metar#radialdist) endpoint in the non-beta API... possibly this also returns a non-sorted list and therefore isn't much different, but it's worked OK till now with a suitable radius value (accepting that sometimes no data will be returned, but mostly just one station is present).  The benefit of [this API](https://www.checkwxapi.com/documentation/metar#metar_latitude_longitude) is that it does the hard work for you in finding the nearest station to a single point.

**Reply by drmrbrewer** [reply]
Fortunately, there seems to be a `dataserver` endpoint in the beta which is 100% compatible with the old ADDS dataserver, at least in terms of input parameters (there are some small differences in the output format of some fieldds, though the individual field names etc are the same... press the dataserver button [on this page](https://beta.aviationweather.gov/data/example/).

For example, compare:

old:  https://www.aviationweather.gov/adds/dataserver_current/httpparam?dataSource=metars&requestType=retrieve&format=xml&radialDistance=50;98.3042759,8.1110951&hoursBeforeNow=1

beta: https://beta.aviationweather.gov/cgi-bin/data/dataserver.php?dataSource=metars&requestType=retrieve&format=xml&radialDistance=50;98.3042759,8.1110951&hoursBeforeNow=1

I'm seeing much better reliability with the beta server... the old ADDS server does sometimes (for me) seem to return wierd/old data.
